src/support/err.c

```diff
@@ -803,3 +842,89 @@ __wt_unexpected_object_type(WT_SESSION_IMPL *session, const char *uri, const cha
 {
     WT_RET_MSG(session, EINVAL, "uri %s doesn't match expected \"%s\"", uri, expect);
 }
+
+/*
+ * __wt_error_log_add --
+ *     Add an entry to the error log.
+ */
+void
+__wt_error_log_add(
+  const char *file, const char *func, int line, const char *expr, int error, int suberror)
+{
+    WT_ERROR_LOG_ENTRY *entry;
+
+    entry = &error_log.log[error_log.tail];
+    entry->file = file;
+    entry->func = func;
+    entry->line = line;
+    entry->expr = expr;
+    entry->error = error;
+    entry->suberror = suberror;
+
+    error_log.count++;
+    error_log.tail = (error_log.tail + 1) % WT_MAX_ERROR_LOG_MAX;
+    if (error_log.head == error_log.tail)
+        error_log.head = (error_log.head + 1) % WT_MAX_ERROR_LOG_MAX;
+}
+
+/*
+ * __wt_error_log_add_ret --
+ *     Add an entry to the error log. Do not return the error code.
+ */
+int
+__wt_error_log_add_ret(
+  const char *file, const char *func, int line, const char *expr, int error, int suberror)
```

Comment:
nit: Do we need this wrapper function? Couldn't the original `__wt_error_log_add()` just return the error and be used directly instead?